### PR TITLE
Split live test runs by service

### PR DESCRIPTION
**Pull request by danieljurek**
The current list of reviewers is probably not complete...

* Use templates to reduce duplications
* Scope variables to relevant projects

For the following projects with live tests: 
* applicationconfig
* batch
* eventhubs
* keyvault
* servicebus
* storage

Once this is merged we'll need to create integration testing pipelines

### CODE REVIEW

**Comment by danieljurek** [comment]
@Azure/azure-sdk-eng

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+parameters:
+  PomFilePath: ''
+  EnvVars: []
+
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml]: nit: extra line

**Comment by weshaggard** [comment]
@mitchdenny do you think it would be best to way until we move the projects before enabling these pipelines? If not it will require that we update all the pipelines as well as the file paths in the yml files after the move.

**Comment by mitchdenny** [comment]
Feel free to merge this now. As I go through and do the restructuring I can add this to the clean-up activities on each one.

**Comment by danieljurek** [comment]
As the repo directory structure changes land we'll have to do two things: 
* Update the tests.yml files to point to the correct archetype yml path 
* Update pipelines to point at the new location for the tests.yml files that have moved

**Pull request merged**